src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -135,51 +134,63 @@ private IEnumerable<MethodBase> FunctionLookup(IList args)
-                            m => m.Name == methodName && m.GetParameters().Length + 1 == args.Count).ToList();
-                    }
+                        var type = asm.GetType(className);
+                        if (type == null) continue;
 
-                    // Check for property getters
-                    if (mi == null || !mi.Any())
-                    {
-                        var prop = type.GetProperties(BindingFlags.Public | BindingFlags.Static | BindingFlags.Instance).Where(
-                                p => p.Name == methodName).FirstOrDefault();
+                        // There should be a way to get the exact method after matching parameter types for a node
+                        // using its function descriptor. AST isn't sufficient for parameter type info.
+                        // Fist check for static methods
+                        mi = type.GetMethods(BindingFlags.Public | BindingFlags.Static).Where(
+                            m => m.Name == methodName && m.GetParameters().Length == args.Count).ToList();
 
-                        if (prop != null)
+                        // Check for instance methods
+                        if (mi == null || !mi.Any())
                         {
-                            mi = prop.GetAccessors().ToList();
+                            mi = type.GetMethods(BindingFlags.Public | BindingFlags.Instance).Where(
+                                m => m.Name == methodName && m.GetParameters().Length + 1 == args.Count).ToList();
                         }
-                    }
 
-                    // TODO: Add check for constructorinfo objects
+                        // Check for property getters
+                        if (mi == null || !mi.Any())
+                        {
+                            var prop = type
+                                .GetProperties(BindingFlags.Public | BindingFlags.Static | BindingFlags.Instance).Where(
+                                    p => p.Name == methodName).FirstOrDefault();
+
+                            if (prop != null)
+                            {
+                                mi = prop.GetAccessors().ToList();
+                            }
+                        }
+
+                        // TODO: Add check for constructorinfo objects
+
+                        if (mi != null)
+                        {
+                            methodCache.Add(key, mi);
+                            break;
+                        }
 
-                    if (mi != null && mi.Any())
+                        //if (method != null)
+                        //{
+                        //    argTypes = method.GetParameters().Select(p => p.ParameterType).ToList();
+                        //    return method.ReturnType;
+                        //}
+                    }
+                }
+                else
+                {
+                    mi = Operators.GetOperatorMethod(methodName);
```

Comment:
does this always return null if it can't find an operator?

Reply:
It currently throws when that happens so not an issue. But I should make this consistent.